fun.lua

```diff
@@ -411,24 +411,45 @@ end
 methods.drop_n = method1(drop_n)
 exports.drop_n = export1(drop_n)
 
+-- Helper of 'wrap_with_elems'
+local gen_with_elems_wrapper = function(gen_param_elems, state)
+    local elems = gen_param_elems[3]
+    if not elems then
+        return gen_param_elems[1](gen_param_elems[2], state)
+    else
+        gen_param_elems[3] = nil
+        return state, unpack(elems)
```

Comment:
How about `{1, nil, 3}` and similar? We often use the `unpack(t, 1, table.maxn(t))` idiom to handle the multireturn values with `nil` in the middle correctly. Can we use it here without a performance loss? I guess, we can, because we execute this code just once for the whole `:drop_while()` iterator.

Reply:
Oh, thanks! I completely forgot about this pitfall of Lua :smile: 
Applied your suggestion, added a test case for multireturn with `nil` for `drop_while` in `tests/stateful.lua`.